### ISSUE TITLE
Corrupt messages from old desktop instance

### ISSUE THREAD

**Issue by michaelkirk-signal** [signalapp/Signal-iOS#4075]
> _Originally posted by @zjalexander in https://github.com/signalapp/Signal-iOS/issues/1415#issuecomment-444589576_

We primarily use the iOS app to talk to each other, no problem.

Within the past 2 days, whenever one of us opens the desktop app (after an admittedly long time of not having the desktop app open) the other immediately gets about 30 "Received message was out of sync" messages.
Desktop log immediately after opening it and my contact got 30 messages: https://debuglogs.org/b769f61ad74762ac2b762a238379c2893ff09f172560236c37705974c6356478
My iOS log https://debuglogs.org/b8df08b9269f31c03f0d1cb75e03f484b478485f50e53554d9959477def072ce.zip

I will try to get my contact to upload their logs as well. 
e: my contact's logs from iOS where they received the "recieved message was out of sync" message https://debuglogs.org/a64c3caa97af7410b515c7f48d3e7a945e054c3c35948cd19fd07bf92da1a7d2.zip

**Comment by michaelkirk-signal** [comment]
@zjalexander - 

Looking at "My iOS log" I don't see any evidence of errors processing messages.

To narrow my search in the above logs, can you clarify, which recipient and devices are seeing the errors, and, to the best of your knowledge, at what time were they received?

**Comment by iceton** [comment]
Having the same problem as of the past few days.
https://debuglogs.org/35be379fe948195c39940c4138a74723eb8d49126f2c080cf783b11eda50c427.zip

**Comment by michaelkirk-signal** [comment]
Hi @iceton - Thanks for the debug log, but we're going to need some more details.

Are *you* seeing "out of sync" messages? Or is the other person? 

Do you have a linked desktop?

Does the other person?

Could you get a debug log from the other person's mobile device and from and involved linked desktop  devices?

Any other details you can provide about what lead up to your experience?

**Comment by iceton** [comment]
@michaelkirk-signal Thanks for your reply! Pardon me for not including -- figured my debug log would contain the relevant info.

As best I can tell, anytime my contact fires up his linked desktop, my iOS goes wild with "Received message was out of sync" messages. Even after resetting the session, this continues. Launching my linked desktop seems to quiet things down a bit, but I still receive a few here and there. I've also received "Error handling incoming message" on my desktop.

Mine: [iOS](https://debuglogs.org/35be379fe948195c39940c4138a74723eb8d49126f2c080cf783b11eda50c427.zip), [Desktop](https://debuglogs.org/1226923cd28b31c1056489687675a24b05aa01e53633b2fa7740c80ef78d9ae0)

Contact: [Android](https://debuglogs.org/d80399cbe40271b3417fce1e8d0742f154c7851e04ddd4bb6f31a236346da6da), [Desktop](https://gist.github.com/iceton/435e79c2eec528925540ab87ecb5f240)

Thanks for your help! Let me know if I can provide any other info.

**Comment by michaelkirk-signal** [comment]
@iceton - The first error I see in your iOS log came in at 2018/12/06 00:51:04:581, but your contact has multiple linked desktops and the desktop log which you added is from one other than this initial error. 

Can you ask them for a log from that device?

**Comment by jonnyflash-owlgod** [comment]
Seem to have the same issue, my contact is using desktop app and I see the multiple out of sync messages

https://debuglogs.org/fbf123a17f6b5aa01ae7f75d2c498131750880413511e978e7d415af8fa4e760.zip

**Comment by zjalexander** [comment]
Happened again to me. My contact opened up Signal Desktop after having it not open. While it was loading previous messages I started getting spammed with the “message received out of sync” error. My contact got frustrated with the loading time of the desktop app and closed it out before I could get them to obtain logs. Here are my ios device logs where I started getting the messages at roughly 8pm https://debuglogs.org/971676befbfd21b5b52c2e8d9c2f28eebf4738823edc7a3b02bbe8d331f6aeee.zip
